### PR TITLE
Issue784 ahu with hydraulic modules

### PR DESCRIPTION
**Pull request by alexanderAKU**
> Thank you for making a Pull Request to AixLib!

> Please consider these points before submitting:
> - Mention the issue that this Pull Request intends to solve, and briefly describe what the main changes are.
> - Make sure all new code follows our [Modelica Guidelines](https://github.com/RWTH-EBC/AixLib/wiki/Modelica-guidelines)
> - Make sure all new models include documentation as described in the [Documentation Wiki-Page](https://github.com/RWTH-EBC/AixLib/wiki/Documentation) and are demonstrated by suitable examples

> If you know someone who could review your code, please use the *Reviewer* function on the right to ask for a review and also assign the PR to them. If not, no worries, we will find someone for you.

> Thanks again for your PR! You can delete these first lines before submitting your PR.

### CODE REVIEW

File: AixLib/Systems/ModularAHU/GenericAHU.mo [hunk 1]
```diff
@@ -0,0 +1,694 @@
...
+  Modelica.Fluid.Interfaces.FluidPort_b port_b1(redeclare package Medium =
+        Medium1,
+    h_outflow(start=Medium1.h_default, nominal=Medium1.h_default))
+    "Fluid connector b1 (positive design flow direction is from port_a1 to port_b1)"
+    annotation (Placement(transformation(extent={{230,-10},{210,10}}),
+        iconTransformation(extent={{232,-10},{212,10}})));
+  Modelica.Fluid.Interfaces.FluidPort_a port_a2(redeclare package Medium =
+        Medium1,
+    h_outflow(start=Medium1.h_default, nominal=Medium1.h_default))
+    "Fluid connector a1 (positive design flow direction is from port_a1 to port_b1)"
```
Comment (KremerMartin) [inline AixLib/Systems/ModularAHU/GenericAHU.mo:60]: Should be a2 here in the comment. ;)

File: AixLib/Systems/ModularAHU/GenericAHU.mo [hunk 2]
```diff
@@ -0,0 +1,694 @@
...
+    annotation (Placement(transformation(extent={{210,70},{230,90}}),
+        iconTransformation(extent={{212,70},{232,90}})));
+  Modelica.Fluid.Interfaces.FluidPort_b port_b2(redeclare package Medium =
+        Medium1,
+    h_outflow(start=Medium1.h_default, nominal=Medium1.h_default))
+    "Fluid connector b1 (positive design flow direction is from port_a1 to port_b1)"
```
Comment (KremerMartin) [inline AixLib/Systems/ModularAHU/GenericAHU.mo:66]: Same here: b2

File: AixLib/Systems/ModularAHU/GenericAHU.mo [hunk 3]
```diff
@@ -0,0 +1,694 @@
...
+    final m1_flow_nominal=m1_flow_nominal,
+    final m2_flow_nominal=m1_flow_nominal,
+    energyDynamics=energyDynamics,
+    massDynamics=massDynamics,
+    final T1_start=T_start,
+    final T2_start=T_start)
+    annotation (Dialog(enable=true, group="Heat recovery heat exchanger"),Placement(transformation(extent={{-20,-10},{-62,42}})));
+  Modelica.Fluid.Interfaces.FluidPort_a port_a3(redeclare package Medium =
+        Medium2) if     usePreheater
+    "Fluid connector a2 (positive design flow direction is from port_a2 to port_b2)"
```
Comment (KremerMartin) [inline AixLib/Systems/ModularAHU/GenericAHU.mo:123]: Same here: a3 instead of a2 (also b3)

**Review by KremerMartin — CHANGES_REQUESTED** [review]
Thanks for the contribution and really good work.
Some Changes need to be provided to the models:

- [x] The Initial Conditions in the Examples `GenericAHU` and `EONERC_AHU2` are not fully specified.

- [ ] Unit-Tests are missing for the examples (also for the old examples of the sub-components)

- [x] Comments are missing in `CtrAHUBasic` for bus-connectors and constants

- [ ] Comments are missing in `CTRRegBasic` for bus-connectors and constants

- [ ] Comments are missing in `GenericAHU` for components and Sensors

- [ ] The Modelica-Guidelines are not fully respected. Please check again for naming-conventions.

- [ ] Just a suggestion: The extract air is abbreviated with `Ret` (for aexample `fanRet`). Since the DIN 13779 provides _Eta_ as abbreviation for the extract air, it would be more applicable to use this abbreviation instead. Same goes for `senTExh`, that would then be `senTEha`.

**Review by KremerMartin — CHANGES_REQUESTED** [review]
When  checking, an error and warnings occurs for the register module:
grafik
Maybe the warnings can be fixed by giving some start values for m_flow_nominal:
grafik


Moreover an error occurs when checking the "CtrAHUBasic" (used in example "GenericAHU" and "EONERC_AHU2"):
grafik

File: AixLib/Systems/ModularAHU/Examples/EONERC_AHU2.mo [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+        rotation=270,
+        origin={-26,-40})));
+  Fluid.Sources.Boundary_pT SourceCooler(
+    nPorts=1,
+    redeclare package Medium = Media.Water,
+    T=283.15) annotation (Placement(transformation(
+        extent={{8,-8},{-8,8}},
+        rotation=270,
+        origin={-2,-40})));
+  Fluid.Sources.Boundary_pT SinkSink(
```
Comment (FelixStege) [inline AixLib/Systems/ModularAHU/Examples/EONERC_AHU2.mo:170]: sinkCooler

File: AixLib/Systems/ModularAHU/Examples/GenericAHU.mo [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        rotation=270,
+        origin={-26,-40})));
+  Fluid.Sources.Boundary_pT SourceCooler(
+    nPorts=1,
+    redeclare package Medium = Media.Water,
+    T=283.15) annotation (Placement(transformation(
+        extent={{8,-8},{-8,8}},
+        rotation=270,
+        origin={-2,-40})));
+  Fluid.Sources.Boundary_pT SinkSink(
```
Comment (FelixStege) [inline AixLib/Systems/ModularAHU/Examples/GenericAHU.mo:120]: sinkCooler

File: AixLib/Systems/ModularAHU/Examples/GenericAHU.mo [hunk 6]
```diff
@@ -0,0 +1,183 @@
+within AixLib.Systems.ModularAHU.Examples;
+model GenericAHU "Example of generic ahu model"
+  extends Modelica.Icons.Example;
+  AixLib.Systems.ModularAHU.GenericAHU genericAHU(
+    redeclare package Medium1 = Media.Air,
+    redeclare package Medium2 = Media.Water,
+    energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial,
```
Comment (FelixStege) [inline AixLib/Systems/ModularAHU/Examples/GenericAHU.mo:7]: setting massDynamics the same way removes the initial value warnings

File: AixLib/Systems/ModularAHU/Examples/EONERC_AHU2.mo [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+  AixLib.Systems.ModularAHU.GenericAHU genericAHU(
+    redeclare package Medium1 = Media.Air,
+    redeclare package Medium2 = Media.Water,
+    T_amb=293.15,
+    m1_flow_nominal=1,
+    m2_flow_nominal=0.5,
+    usePreheater=true,
+    useHumidifierRet=false,
+    useHumidifier=true,
+    energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial,
```
Comment (FelixStege) [inline AixLib/Systems/ModularAHU/Examples/EONERC_AHU2.mo:14]: setting massDynamics the same way removes the initial value warnings

**Review by FelixStege — CHANGES_REQUESTED** [review]
-  Checking the model AixLib.Systems.ModularAHU.RegisterModule produces errors
RegisterModule

- Simulation of AixLib.Systems.ModularAHU.Examples.GenericAHU and AixLib.Systems.ModularAHU.Examples.EONERC_AHU2 produces warnings:
GenericAHU

- Simulation of AixLib.Systems.ModularAHU.Validation.AHU2_Cooler, AHU2_Heater and AHU2_Preheater shows a warning, which can be removed by adding a constant boolean signal
AHU2_Cooler

File: AixLib/Systems/ModularAHU/GenericAHU.mo [hunk 8]
```diff
@@ -0,0 +1,708 @@
...
+  parameter Fluid.Movers.BaseClasses.Characteristics.efficiencyParameters
+    hydraulicEfficiency(
+      V_flow={0},
+      eta={0.7}) "Hydraulic efficiency of the fans" annotation (dialog(group="Fans"));
+
+  parameter  Modelica.SIunits.Temperature T_amb "Ambient temperature";
+  parameter Modelica.SIunits.MassFlowRate m1_flow_nominal(min=0)
+    "Nominal mass flow rate in air canal";
+  parameter Modelica.SIunits.MassFlowRate m2_flow_nominal(min=0)
+    "Nominal mass flow rate in hydraulics";
```
[the remark this answers is not in the document]
Comment (alexanderAKU) [reply AixLib/Systems/ModularAHU/GenericAHU.mo:36]: min modifier deltetd

File: AixLib/Systems/ModularAHU/GenericAHU.mo [hunk 9]
```diff
@@ -0,0 +1,708 @@
...
+    "Fluid connector a2 (positive design flow direction is from port_a2 to port_b2)"
+    annotation (Placement(transformation(extent={{210,70},{230,90}}),
+        iconTransformation(extent={{212,70},{232,90}})));
+  Modelica.Fluid.Interfaces.FluidPort_b port_b2(redeclare package Medium =
+        Medium1,
+    h_outflow(start=Medium1.h_default, nominal=Medium1.h_default))
+    "Fluid connector b2 (positive design flow direction is from port_a2 to port_b2)"
+    annotation (Placement(transformation(extent={{-210,70},{-230,90}}),
+        iconTransformation(extent={{-210,70},{-230,90}})));
+  RegisterModule perheater(
```
[the remark this answers is not in the document]
Comment (alexanderAKU) [reply AixLib/Systems/ModularAHU/GenericAHU.mo:72]: done

File: AixLib/Systems/ModularAHU/Controller/CtrAHUBasic.mo [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+    final yMin=0,
+    final controllerType=Modelica.Blocks.Types.SimpleController.PID,
+    final k=k,
+    final Ti=Ti,
+    final Td=0,
+    final initType=initType,
+    y_start=y_start,
+    final reverseAction=false,
+    final reset=AixLib.Types.Reset.Disabled) if useTwoFanCtr
+    "PID controller for retrun fan. Is deactivated if useTwoFanCtr is false."
```
[the remark this answers is not in the document]
Comment (alexanderAKU) [reply AixLib/Systems/ModularAHU/Controller/CtrAHUBasic.mo:93]: done

File: AixLib/Systems/ModularAHU/BaseClasses/GenericAHUBus.mo [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+  "Data bus for generic air-handling unit"
+  extends Modelica.Icons.SignalBus;
+  import SI = Modelica.SIunits;
+  Systems.ModularAHU.BaseClasses.RegisterBus preheaterBus;
+  Systems.ModularAHU.BaseClasses.RegisterBus coolerBus;
+  Systems.ModularAHU.BaseClasses.RegisterBus heaterBus;
+  SI.Temperature TOutsAirMea "Outside air temperature";
+  SI.Temperature TSupAirMea "Supply air temperature";
+  SI.Temperature TRetAirMea "Return air temperature";
+  SI.Temperature TExhAirMea "Return air temperature";
```
[the remark this answers is not in the document]
Comment (alexanderAKU) [reply AixLib/Systems/ModularAHU/BaseClasses/GenericAHUBus.mo:12]: done

File: AixLib/Systems/ModularAHU/Examples/EONERC_AHU2.mo (hunk 4 above)
Comment (alexanderAKU) [reply AixLib/Systems/ModularAHU/Examples/EONERC_AHU2.mo:170]: done

File: AixLib/Systems/ModularAHU/Examples/GenericAHU.mo (hunk 5 above)
Comment (alexanderAKU) [reply AixLib/Systems/ModularAHU/Examples/GenericAHU.mo:120]: done

File: AixLib/Systems/ModularAHU/Examples/GenericAHU.mo (hunk 6 above)
Comment (alexanderAKU) [reply AixLib/Systems/ModularAHU/Examples/GenericAHU.mo:7]: done

File: AixLib/Systems/ModularAHU/Examples/EONERC_AHU2.mo (hunk 7 above)
Comment (alexanderAKU) [reply AixLib/Systems/ModularAHU/Examples/EONERC_AHU2.mo:14]: Done

**Review by KremerMartin — APPROVED** [review]

**Review by FelixStege — APPROVED** [review]

**Pull request merged**